internal/services/containers/kubernetes_cluster_node_pool_resource.go

```diff
@@ -435,6 +442,17 @@ func resourceKubernetesClusterNodePoolCreate(d *pluginsdk.ResourceData, meta int
 		return err
 	}
 
+	subnetID, err := commonids.ParseSubnetID(d.Get("vnet_subnet_id").(string))
```

Comment:
SubnetID isn't always going to be specified so we should probably wrap all of this logic in a check 
```suggestion
	if subnetId, ok := d.GetOk("vnet_subnet_id").(string); ok && subnetId != "" {
	...
	}
```

Reply:
gotcha, makes sense!